File: arrPrime.cpp
#include <iostream>

struct List {
	int val;
	List *next;
};

bool isPrime(int x) {

	if (x == 2)
		return true;

	if (x < 2 || x % 2 == 0)
		return false;
	for (int d = 3; d * d <= x; d += 2)
		if (x % d == 0)
			return false;

	return true;
}

void readList(List *&f, int len) {

	f = new List;
	std::cout << "val...\n";
	std::cin >> f->val;
	f->next = 0;

	if (len > 1) {

		List *l = f;
		for (int i = 1; i < len; i++) {

			List *p = new List;
			p->next = 0;
			std::cout << "val...\n";
			std::cin >> p->val;

			l->next = p;
			l = p;
		}
	}
}

void prntList(List *f) {

	for (List *p = f; p; p = p->next)
		std::cout << p->val << ' ';

	std::cout << "\n";
}

void prntIfPrime(List *f) {

	for (List *p = f; p; p = p->next)
		if (isPrime(p->val))
			std::cout << p->val << " is prime\n";
		else
			std::cout << p->val << " is not prime\n";
}

int main() {

	List *f = 0;
	int len;

	std::cout << "Lenght...\n";
	std::cin >> len;
	readList(f, len);

	std::cout << "Read list is:\n";
	prntList(f);

	std::cout << "The prime elements are:\n";
	prntIfPrime(f);

	return 0;
}

File: arrPrime.py
def readList(mList, leng):

	x = int(input("new element..."))
	mList[0] = x

	for i in range(1, leng):
		x = int(input("new element..."))
		mList.append(x)

def isPrime(x):
	if x == 2:
		return True
	if x < 2 or x % 2 == 0:
		return False

	d = 3
	while d * d <= x:
		if x % d == 0:
			return False
		d += 2

	return True

def prntIfPrime(mList):
	for i in range(0, len(mList)):
		print mList[i]
		if isPrime(mList[i]):
			print "is prime"
		else:
			print "is not prime"

mList = [0]
leng = int(input("Lenght of list..."))
readList(mList, leng)

print "Read list is:"
print mList

print "The prime elements are:"
prntIfPrime(mList)